wordpress-mobile/gutenberg-mobile
# iOS - Add ability to scroll to caret when typing goes under the keyboard

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/464

This PR adds ability to follow caret when typing goes under the keyboard. 

![caret-follow3](https://user-images.githubusercontent.com/5032900/51181451-3e621d80-18dc-11e9-91b9-b6675e124587.gif)

![caret-follow4](https://user-images.githubusercontent.com/5032900/51181503-62256380-18dc-11e9-8a46-fe050e4ddd43.gif)

WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/10826

Depends on PRs:
https://github.com/wordpress-mobile/react-native-aztec/pull/112
https://github.com/WordPress/gutenberg/pull/13323
https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/3

**To Test**

Prerequisites

- Run `yarn install` <-- important `yarn start` `yarn ios`
- Might need to remember the scrolling mechanism, reviewers can refer to the diagram from [this PR](https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/2).
 
Test 1 - Paragraph block - Add new text to end
- Add new text at the end of the block until caret goes under the keyboard
- Verify that it automatically refreshes the scroll position to show the caret and also the block toolbar

Note: it is also possible to scroll for only showing the caret without the block toolbar, I wanted to keep this consistent with what happens when we open the keyboard, however, we can still change it for typing case. 

Test 2 - Paragraph block - Add new text in the middle
- Add new text in the middle of the block until caret goes under the keyboard
- Verify that it automatically refreshes the scroll position to show the caret

Test 3 - Repeat tests 1,2 for Heading block

Test 4 - Android

We don't expect any change on Android side.

- Run `yarn android`
- Make sure Android example app opens successfully and you can do basic operations on it
